### PR TITLE
Added oGCD tracking for SMN.

### PR DESCRIPTION
**Pull request by KelosPelmand**
This pull should be handled after #474

### CODE REVIEW

File: src/data/ACTIONS/SMN.js [hunk 1]
```diff
@@ -31,13 +42,15 @@ export default {
...
 		cooldown: 30,
+		cooldownGroup: SMN_COOLDOWN_GROUP.ENERGY,
 	},
 
 	ENKINDLE: {
 		id: 184,
 		name: 'Enkindle',
 		icon: 'https://xivapi.com/i/002000/002677.png',
 		cooldown: 120,
+		cooldownGroup: SMN_COOLDOWN_GROUP.ENKINDLE, //this skill isn't techically usable anymore, but would be part of this group
```
Comment (ackwell) [inline src/data/ACTIONS/SMN.js]: If the skill isn't usable, probably better off removing the data - there isn't a reason to keep it sitting around.

File: src/data/ACTIONS/ACN.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ export default {
 		name: 'Summon',
 		icon: 'https://xivapi.com/i/000000/000516.png',
 		cooldown: 10,
+		cooldownGroup: 180, //SMN_COOLDOWN_GROUP.SUMMON
```
Comment (ackwell) [inline src/data/ACTIONS/ACN.js]: are you able to import+use the commented reference?

File: src/parser/jobs/smn/modules/GeneralCDDowntime.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import ACTIONS from 'data/ACTIONS'
+import CooldownDowntime from 'parser/core/modules/CooldownDowntime'
+
+export default class GeneralCDDowntime extends CooldownDowntime {
+	//Time that Jump deems ok for a OGCD to be down : ^)
```
Comment (ackwell) [inline src/parser/jobs/smn/modules/GeneralCDDowntime.js]: jump does smn stuff? :woah:

File: src/parser/core/modules/CooldownDowntime.js [hunk 4]
```diff
@@ -43,27 +43,30 @@ export default class CooldownDowntime extends Module {
...
-						id,
-						encounterLength - timeOnCooldown + allowedDowntime,
-						firstUseOffset,
-						encounterLength
-					),
-				})
-			)
+		const groups = {}
+		this.trackedCds.forEach(id  => {
+			const groupId = getDataBy(ACTIONS, 'id', id).cooldownGroup || 0
```
Comment (ackwell) [inline src/parser/core/modules/CooldownDowntime.js]: This has the theoretical potential to error, if the ID is ever not found. Probably _ok_ in this situation, but avoidable.

I'd rather use something like `'unknown'` or `'ungrouped'` for the fallback, though - `0` is a bit of a magic value.

File: src/parser/core/modules/CooldownDowntime.js [hunk 5]
```diff
@@ -43,27 +43,30 @@ export default class CooldownDowntime extends Module {
...
+		for (const key in groups) {
+			if (key === '0') {
+				//handle non-grouped cooldowns
+				groups[key].map(id => {
+					OGCDRequirements.push(this.checkCooldown(id, <ActionLink {...getDataBy(ACTIONS, 'id', id)} />, encounterLength))
+				})
+			} else {
+				//handle cooldown group
+				const displayName = []
+				groups[key].forEach(id => {
```
Comment (ackwell) [inline src/parser/core/modules/CooldownDowntime.js]: Can simplify this to just a `.map`

File: src/parser/jobs/smn/modules/GeneralCDDowntime.js (hunk 3 above)
Comment (KelosPelmand) [reply src/parser/jobs/smn/modules/GeneralCDDowntime.js]: I copy/pasted that file from RDM and adjusted the skill list.  I just left the note as what little "where this number came from" as was there.

**Pull request merged**